api_backend.yaml

```diff
@@ -261,6 +261,54 @@ paths:
           description: There was an error in retrieving the message.
           schema:
             $ref: "#/definitions/ProblemJson"
+  "/messages/{id}/message-status":
+    x-swagger-router-controller: MessagesController          
+    put:
+      operationId: upsertMessageStatusAttributes
+      summary: UpsertMessageStatusAttributes
+      description: Updates the status of a message with attributes
+      parameters:
+        - name: id
+          in: path
+          type: string
+          required: true
+          description: The ID of the message.
+        - name: body
+          in: body
+          schema:
+            $ref: "#/definitions/MessageStatusChange"
+          required: true
+          x-examples:
+            application/json: |
+              change_type: "bulk",
+              is_archived: true,
+              is_read: true
+      responses:
+        "200":
```

Comment:
Nitpick: 

`204 No Content` is more appropriate as we're returning nothing (even `205 Reset` would make the case, although is a bit more exotic)